control-plane-pki-operator/operator.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	hypershiftv1beta1 "github.com/openshift/hypershift/client/clientset/clientset/typed/hypershift/v1beta1"
+	"github.com/openshift/hypershift/control-plane-pki-operator/certrotationcontroller"
+	"github.com/openshift/hypershift/control-plane-pki-operator/targetconfigcontroller"
+	"github.com/openshift/library-go/pkg/controller/controllercmd"
+	"github.com/openshift/library-go/pkg/operator/certrotation"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+func RunOperator(ctx context.Context, controllerContext *controllercmd.ControllerContext) error {
+	var namespace, name string
+	for env, target := range map[string]*string{
+		"HOSTED_CONTROL_PLANE_NAMESPACE": &namespace,
+		"HOSTED_CONTROL_PLANE_NAME":      &name,
+	} {
```

Comment:
fwiw there must only be a single HCP so onlny knowing the naespace is stricly required

Reply:
Yeah, I actually got hung up on this when I was learning the code-base ... if it's required to be a singleton, let's not write code that happens to look like it might work fine if there are N in the namespace - like the CPO ;)